### ISSUE TITLE
Parser optionalVars.hasMore in models

### ISSUE THREAD

**Issue by ybelenko** [OpenAPITools/openapi-generator#710]
##### Description

Found weird parser bug.
Last variable in `optionalVars` array shows `hasMore: true` which is obviously wrong.
It's even more weird that this behavior noticed only with `FormatTest`(definition: `format_test`) model withing all spec.
This bug breaks constructor function in models generation.
Mustache
```php
public function construct(
    {{#hasRequired}}
    {{#requiredVars}}
    ${{name}}{{#hasMore}},{{/hasMore}}{{^hasMore}}{{#hasOptional}},{{/hasOptional}}{{/hasMore}}
    {{/requiredVars}}
    {{/hasRequired}}
    {{#hasOptional}}
    {{#optionalVars}}
    ${{name}} = null{{#hasMore}},{{/hasMore}}
    {{/optionalVars}}
    {{/hasOptional}}
) {

}
```
Output:
```php
public function __construct(
    $number,
    $byte,
    $date,
    $password,
    $integer = null,
    $int32 = null,
    $int64 = null,
    $float = null,
    $double = null,
    $string = null,
    $binary = null,
    $dateTime = null,
    $uuid = null,
) {

}
```

Link to parser output:
https://gist.github.com/ybelenko/89f11291565ed4f8719061c89f082f39#file-formattest-json-L1296-L1345
```json
{
  "optionalVars": [
    {
      "baseName": "uuid",
      "getter": "getUuid",
      "setter": "setUuid",
      "dataType": "string",
      "datatypeWithEnum": "string",
      "dataFormat": "uuid",
      "name": "uuid",
      "defaultValueWithParam": " = data.uuid;",
      "baseType": "string",
      "jsonSchema":
        "{\r\n  \"type\" : \"string\",\r\n  \"format\" : \"uuid\"\r\n}",
      "exclusiveMinimum": false,
      "exclusiveMaximum": false,
      "hasMore": true,
      "required": false,
      "secondaryParam": false,
      "hasMoreNonReadOnly": true,
      "isPrimitiveType": true,
      "isContainer": false,
      "isNotContainer": true,
      "isString": true,
      "isNumeric": false,
      "isInteger": false,
      "isLong": false,
      "isNumber": false,
      "isFloat": false,
      "isDouble": false,
      "isByteArray": false,
      "isBinary": false,
      "isFile": false,
      "isBoolean": false,
      "isDate": false,
      "isDateTime": false,
      "isUuid": true,
      "isListContainer": false,
      "isMapContainer": false,
      "isEnum": false,
      "isReadOnly": false,
      "vendorExtensions": {},
      "hasValidation": false,
      "isInherited": false,
      "nameInCamelCase": "Uuid",
      "nameInSnakeCase": "UUID",
      "isXmlAttribute": false,
      "isXmlWrapped": false,
      "iexclusiveMaximum": false,
      "datatype": "string"
    }
  ]
}
```

##### openapi-generator version

3.2.0-SNAPSHOT

##### OpenAPI declaration file content or url

https://github.com/OpenAPITools/openapi-generator/blob/ab08dd1d26c3140f35fe9e0b65231fdde27de9ce/modules/openapi-generator/src/test/resources/2_0/petstore-with-fake-endpoints-models-for-testing.yaml#L1221-L1277

##### Command line used for generation

```shell
$ bin/php-slim-server-petstore.sh
``` 
with `-DdebugModels` option

##### Steps to reproduce

Run generate execution with `-DebugModels` option and point to fake spec from description above

**Comment by wing328** [comment]
@ybelenko thanks for reporting the issue. What about using `{{#-first}}` or `{{#-last}}` as a workaround for the time being?

**Comment by ybelenko** [comment]
Yeah, I've checked it locally, this bug fixed in #1360
